packages/core/src/utils.ts

```diff
@@ -0,0 +1,240 @@
+    const typeQuery = options?.routeTypes ? `?resourceType=${options.routeTypes.join(',')}` : '';
+    const url = `${indexerUrl}/api/routes/from/${domainConfig.id}${typeQuery}`;
+    const response = await fetch(url);
+    const data = (await response.json()) as IndexerRoutesResponse;
+
+    let routeFeeHandlerAddressesAndTypes: Array<
+      RouteIndexerType & { feeHandlerAddress: string } & { feeHandlerType: FeeHandlerType }
+    >;
+
+    if (domainConfig.type === Network.EVM && options?.sourceProvider) {
+      const routesWithHandlerAddresses = await getFeeHandlerAddressesOfRoutes({
+        routes: data.routes,
+        chainId: domainConfig.chainId,
+        bridgeAddress: domainConfig.bridge,
+        provider: options.sourceProvider,
+      });
+
+      routeFeeHandlerAddressesAndTypes = await getFeeHandlerTypeOfRoutes({
+        routes: routesWithHandlerAddresses,
+        chainId: domainConfig.chainId,
+        provider: options.sourceProvider,
+      });
+    }
+
+    return data.routes.map(route => {
+      const resource = domainConfig.resources.find(
+        resource => resource.resourceId === route.resourceId,
+      )!;
+
+      let routeWithTypeAndAddress;
+      if (routeFeeHandlerAddressesAndTypes) {
+        routeWithTypeAndAddress = routeFeeHandlerAddressesAndTypes.find(_route => {
+          _route.fromDomainId === route.fromDomainId &&
+            _route.toDomainId === route.toDomainId &&
+            _route.resourceId === route.resourceId;
+        });
+      }
+
+      let feeHandler = undefined;
+      if (routeWithTypeAndAddress) {
+        feeHandler = {
+          type: routeWithTypeAndAddress.feeHandlerType,
+          address: routeWithTypeAndAddress.feeHandlerAddress,
+        };
+      }
+
+      return {
+        fromDomain: config.getDomain(domainConfig.chainId),
+        toDomain: config.findDomainConfigBySygmaId(Number(route.toDomainId)),
+        resource: resource,
+        feeHandler,
+      };
+    });
+  } catch (err) {
+    if (err instanceof Error) {
+      throw new Error(`Failed to fetch routes because of: ${err.message}`);
+    } else {
+      throw new Error('Something went wrong while fetching routes');
+    }
+  }
+}
+
+/**
+ * TODO: why isn't txHash unique identifier
+ * @param txHash
+ */
+export async function getTransferStatus(
+  txHash: string,
+  environment: Environment = process.env.SYGMA_ENV,
+): Promise<TransferStatusResponse> {
+  const env = environment ?? Environment.MAINNET;
+  const { url, explorerUrl } = getIndexerTransferUrl(env, txHash);
+
+  try {
+    const response = await fetch(url);
+    const data = (await response.json()) as (Record<string, unknown> & {
+      status: TransferStatus;
+      toDomainId: number;
+      fromDomainId: number;
+      sourceHash: string;
+      destinationHash: string;
+      depositNonce: number;
+    })[];
+
+    if (!data.length) throw new Error(`Record for ${txHash} not found`);
+
+    return {
+      status: data[0].status,
```

Comment:
would it make sense to destrucuture the array here, since in any case we are just using the first element?